### PR TITLE
refactor(_tools): use `fs/walk` in deprecations check

### PR DESCRIPTION
**Pull request by iuioiua**
This PR also adds some internal directories to the ignored directory list, making the check quicker.

### CODE REVIEW

File: _tools/check_deprecation.ts [hunk 1]
```diff
@@ -51,87 +57,76 @@ const DEFAULT_DEPRECATED_VERSION = semver.inc(
...
-      continue;
-    }
+for await (
+  const { path } of walk(ROOT, {
+    includeDirs: false,
+    exts: EXTENSIONS,
+    skip: EXCLUDED_PATHS.map((path) => new RegExp(path + "$")),
+  })
+) {
+  console.log(path);
```
Comment (kt3k) [inline _tools/check_deprecation.ts:67]: Is this for debugging?

**Review by kt3k — COMMENTED** [review]
Nice! Left one question

File: _tools/check_deprecation.ts (hunk 1 above)
Comment (iuioiua) [reply _tools/check_deprecation.ts:67]: Oh! Yes, it is. I'll fix.

**Review by kt3k — APPROVED** [review]
LGTM 👍

**Pull request merged**